CHANGELOG.md

```diff
@@ -6,9 +6,156 @@
 
 ## v4.1
 
-### 4.1.0 - Highlights
+### Custom CSS
 
-#### Moved plugins
+Jupyterlab now supports automatic loading of custom CSS.
+Themes are the recommended way for customizing the JupyterLab look and feel,
+while custom CSS is intended for minor personal adjustments.
+
+To opt-in start jupyterlab with `--custom-css` flag.
+The location of the `custom.css` file is documented in the section on [customizing the user interface](https://jupyterlab.readthedocs.io/en/latest/user/interface_customization.html#custom-css).
+
+```{note}
+`custom.css` is also available in Jupyter Notebook;
+while similar selectors can be used for components shared between JupyterLab 4.1 and Jupyter Notebook 7.0,
+a different set of selectors was used in the Notebook 6 and older.
+Users with the custom CSS file using the selectors from legacy Notebook versions
+will need to update them and note that tutorials may refer to the selectors for the old Notebook versions.
+```
+
+### Diagrams in Markdown
+
+Matching GitHub-Flavoured Markdown, JupyterLab 4.1 now supports [Mermaid](https://github.com/mermaid-js/mermaid) diagrams.
+To create a mermaid diagram use the `mermaid` language specifier for a code block in a markdown cell or document, for example:
+
+~~~
+```mermaid
+flowchart LR
+
+A[Hard] -->|Text| B(Round)
+B --> C{Decision}
+C -->|One| D[Result 1]
+C -->|Two| E[Result 2]
+```
+~~~
+
+which renders as:
+
+<img alt="Rendered Mermaid diagram going from left to right with modern look; the diagram contains blueish square node with text Hard pointing to round square node with text Round; over the arrow connecting the two nodes there is a word Text; the second node further connects to a rhombus-shaped node with text Decision which then connects to two further nodes, Result 1 and Result 2." src="https://raw.githubusercontent.com/krassowski/jupyterlab/changelog-4.1/docs/source/getting_started/changelog_assets/4.1-merimad-diagram.png" class="jp-screenshot">
+
+
+### Inline Completer
+
+JupyterLab now supports completion presented as ghost text in the cell and file editors,
+enabling multi-line completions such as those generated by large language models (like GPT-based copilots).
```

Comment:
```suggestion
allowing generative AI models to provide multi-line completions.
```

OpenAI asserts ownership of the name "GPT", and "CoPilot" is a trademark of Microsoft.

Reply:
But copilot lowercase might be ok? Do you have a source on GPT being a trademark?

Reply:
"Copilot" has been used for ~100 years in aviation, but in generative AI terms, it seems vendor-specific. https://openai.com/brand says, 'The "OpenAI" name, the OpenAI logo, the "ChatGPT" and “GPT” brands, and other OpenAI trademarks, are property of OpenAI.' OpenAI's [GPT trademark application](https://tsdr.uspto.gov/#caseNumber=97733259&caseSearchType=US_APPLICATION&caseType=DEFAULT&searchType=statusSearch) was filed on December 27, 2022, and is "under examination" today. I'm biased as an employee of a company that makes large language models, but I don't think any one company's LLM-related intellectual property (irrespective of trademark approval status) should be in Jupyter's docs.

Reply:
> This is why in a previous iteration I suggested "GPT-based copilots" over "GitHub Copilot" suggested by Jeremy)

Yes, the original suggestion was to have the word "copilot" in the changelog, because this is what most users have heard of, and also what they used when requesting the feature: https://github.com/jupyterlab/jupyterlab/issues/14267

The high number of :+1: suggests folks come to the repo and search for "copilot":

![image](https://github.com/jupyterlab/jupyterlab/assets/591645/dc554fb8-0e97-44dd-8842-afd6b485e2e3)

Reply:
Thanks for the reference on "GPT" trademark claim from OpenAI. It is interesting to see if their application will be accepted, especially because I would argue this is a generic term for a family of models. For anyone interested further, there is a section in Wiki on this: https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#Brand_issues

> Relatedly, OpenAI has applied to the [United States Patent and Trademark Office](https://en.wikipedia.org/wiki/United_States_Patent_and_Trademark_Office) (USPTO) to seek domestic [trademark registration](https://en.wikipedia.org/wiki/Trademark_registration) for the term “GPT” in the field of AI.[[71]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-:5-71) OpenAI sought to expedite handling of its application, but the USPTO declined that request in April 2023.[[75]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-:6-75) In May 2023, the USPTO responded to the application with a determination that "GPT" was both descriptive and generic.[[76]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-76) As of November 2023, OpenAI continues to pursue its argument through the available processes. Regardless, failure to obtain a registered U.S. trademark does not preclude some level of common-law trademark rights in the U.S.,[[77]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-77) and/or trademark rights in other countries.[[78]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-78)
>
> For any given type or scope of trademark protection in the U.S., OpenAI would need to establish that the term is actually “[distinctive](https://en.wikipedia.org/wiki/Trademark_distinctiveness)” to their specific offerings in addition to being a broader technical term for the kind of technology. Some media reports suggested that OpenAI may be able to obtain trademark registration based indirectly on the fame of its GPT-based [chatbot](https://en.wikipedia.org/wiki/Chatbot) product, [ChatGPT](https://en.wikipedia.org/wiki/ChatGPT),[[75]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-:6-75)[[79]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-79) for which OpenAI has separately sought protection (and which it has sought to enforce more strongly).[[80]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-80) **Other reports have indicated that registration for the bare term “GPT” seems unlikely to be granted,[[71]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-:5-71)[[81]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-:7-81) as it is used frequently as a common term to refer simply to AI systems that involve generative pre-trained transformers.[[3]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-:4-3)[[82]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-82)[[83]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-83)[[84]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-84)** In any event, to whatever extent exclusive rights in the term may occur the U.S., others would need to avoid using it for similar products or services in ways likely to cause confusion.[[81]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-:7-81)[[85]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-85) If such rights ever became broad enough to implicate other well-established uses in the field, the trademark doctrine of descriptive fair use could still preserve some room to continue non-brand-related usage.[[86]](https://en.wikipedia.org/wiki/Generative_pre-trained_transformer#cite_note-86)


But yeah, happy to not include this in here if this raises an objection :)

> I'm biased as an employee of a company that makes large language models, but I don't think any one company's LLM-related intellectual property (irrespective of trademark approval status) should be in Jupyter's docs.

I believe we agree on this point, the question is whether this particular term represents "any one company's LLM-related intellectual property" or just a generic term as introduced to scholarly community by scholarly means and simultaneously developed into a commercial product (ChatGPT).